homeassistant/components/cloud/__init__.py

```diff
@@ -133,189 +130,51 @@ def is_cloudhook_request(request):
-            conf = self.google_actions_user_conf
-
-            def should_expose(entity):
-                """If an entity should be exposed."""
-                if entity.entity_id in CLOUD_NEVER_EXPOSED_ENTITIES:
-                    return False
-
-                return conf['filter'](entity.entity_id)
-
-            self._gactions_config = ga_h.Config(
-                should_expose=should_expose,
-                allow_unlock=self.prefs.google_allow_unlock,
-                entity_config=conf.get(CONF_ENTITY_CONFIG),
-            )
-
-        return self._gactions_config
-
-    def path(self, *parts):
-        """Get config path inside cloud dir.
-
-        Async friendly.
-        """
-        return self.hass.config.path(CONFIG_DIR, *parts)
-
-    async def fetch_subscription_info(self):
-        """Fetch subscription info."""
-        await self.hass.async_add_executor_job(auth_api.check_token, self)
-        websession = self.hass.helpers.aiohttp_client.async_get_clientsession()
-        return await websession.get(
-            self.subscription_info_url, headers={
-                'authorization': self.id_token
-            })
-
-    async def logout(self):
-        """Close connection and remove all credentials."""
-        await self.iot.disconnect()
-
-        self.id_token = None
-        self.access_token = None
-        self.refresh_token = None
-        self._gactions_config = None
-
-        await self.hass.async_add_job(
-            lambda: os.remove(self.user_info_path))
-
-    def write_user_info(self):
-        """Write user info to a file."""
-        with open(self.user_info_path, 'wt') as file:
-            file.write(json.dumps({
-                'id_token': self.id_token,
-                'access_token': self.access_token,
-                'refresh_token': self.refresh_token,
-            }, indent=4))
-
-    async def async_start(self, _):
-        """Start the cloud component."""
-        def load_config():
-            """Load config."""
-            # Ensure config dir exists
-            path = self.hass.config.path(CONFIG_DIR)
-            if not os.path.isdir(path):
-                os.mkdir(path)
-
-            user_info = self.user_info_path
-            if not os.path.isfile(user_info):
-                return None
-
-            with open(user_info, 'rt') as file:
-                return json.loads(file.read())
-
-        info = await self.hass.async_add_job(load_config)
-        await self.prefs.async_initialize()
-
-        if info is None:
-            return
-
-        self.id_token = info['id_token']
-        self.access_token = info['access_token']
-        self.refresh_token = info['refresh_token']
-
-        self.hass.async_create_task(self.iot.connect())
-
-    def _decode_claims(self, token):  # pylint: disable=no-self-use
-        """Decode the claims in a token."""
-        from jose import jwt
-        return jwt.get_unverified_claims(token)
+    @callback
```

Comment:
Let's make this async and await the tasks, that way we:
 - can call it with blocking=True and reconnect is done when service call done
 - stack traces are attributed correctly

Reply:
Yeah make sense